client-react/src/pages/app/change-app-plan/ChangeAppPlanDestinationPlanDetails.tsx

```diff
@@ -219,7 +219,7 @@ export const DestinationPlanDetails: React.FC<DestinationPlanDetailsProps> = ({
     }
 
     return options;
-  }, [formProps.values.serverFarmInfo.newPlanInfo?.name, skuTier, serverFarms, t]);
+  }, [skuTier, serverFarms, t]);

```

Comment:
Add missing dependencies.
- `formProps.values.serverFarmInfo.newPlanInfo.name` (lines 214-215)
- `isConsumptionToPremiumEnabled` (line 207)

```suggestion
  }, [formProps.values.serverFarmInfo.newPlanInfo.name, isConsumptionToPremiumEnabled, skuTier, serverFarms, t]);
```

Reply:
The name shouldn't be a dependency. It was causing some errors where the server farms were being updated when it didn't need to be.

Reply:
It is a dependency. However, it happens to work since the `options.length` check ensures that it only runs once. It also coincidentally ensures that the new entry can only be added once and with the first new name it had.. There's a way to do this without fibbing in the dependency list, which is a potential source of bugs. However, for expediency's sake, ignore this bug for now.